### PR TITLE
Fix PUT collections

### PR DESCRIPTION
**Pull request by avbentem**
**Related Issue(s):**

- #665

**Description:**

* Change the registration of the PUT `/collections/{collection_id}` endpoint, to include the missing URL fragment when calling the `update_collection` function
* Add missing tests, and change function signatures in dummy clients to match `BaseTransactionsClient`
* Minor documentation fixes

**PR Checklist:**

- [x] `pre-commit` hooks pass locally
- [x] Tests pass (run `make test`)
- [x] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/main/CHANGES.md).

### CODE REVIEW

**Review by jonhealy1 — APPROVED** [review]
LGTM

File: stac_fastapi/extensions/stac_fastapi/extensions/core/transaction.py [hunk 1]
```diff
@@ -31,6 +31,13 @@ class PutItem(ItemUri):
     item: stac_types.Item = attr.ib(default=Body(None))
 
 
+@attr.s
+class PutCollection(CollectionUri):
+    """Update Collection."""
+
+    collection: stac_types.Collection = attr.ib(default=Body(None))
```
Comment (vincentsarago) [inline stac_fastapi/extensions/stac_fastapi/extensions/core/transaction.py:38]: I'm not sure to understand why we default to `Body(None)` here

Comment (jonhealy1) [reply stac_fastapi/extensions/stac_fastapi/extensions/core/transaction.py:38]: The PutItem class above does the same thing which is maybe why it was done here?

**Review by vincentsarago — APPROVED** [review]

File: stac_fastapi/extensions/stac_fastapi/extensions/core/transaction.py (hunk 1 above)
Comment (avbentem) [reply stac_fastapi/extensions/stac_fastapi/extensions/core/transaction.py:38]: Indeed, I did not give that any thought whatsoever. 🙈

**Pull request merged**